macros/src/lib.rs

```diff
@@ -22,6 +22,16 @@ extern crate mac;
 
 use rustc::plugin::Registry;
 
+macro_rules! panictry {
+    ($e:expr) => ({
+        use syntax::diagnostic::FatalError;
+        match $e {
+            Ok(e) => e,
+            Err(FatalError) => panic!(FatalError)
+        }
+    })
+}
+
```

Comment:
Ah, can you add a link or reference to that libsyntax code in a comment here? And, is there an associated Rust issue number?